homeassistant/components/ubee/device_tracker.py

```diff
@@ -9,79 +14,58 @@
     CONF_HOST, CONF_PASSWORD, CONF_USERNAME)
 import homeassistant.helpers.config_validation as cv
 
-REQUIREMENTS = ['pyubee==0.2']
+REQUIREMENTS = ['pyubee==0.5']
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_MODEL = 'model'
+DEFAULT_MODEL = 'detect'
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_HOST): cv.string,
     vol.Required(CONF_PASSWORD): cv.string,
     vol.Required(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_MODEL, default=DEFAULT_MODEL): cv.string
 })
 
 
 def get_scanner(hass, config):
     """Validate the configuration and return a Ubee scanner."""
-    try:
-        return UbeeDeviceScanner(config[DOMAIN])
-    except ConnectionError:
+    info = config[DOMAIN]
+    host = info[CONF_HOST]
+    username = info[CONF_USERNAME]
+    password = info[CONF_PASSWORD]
+    model = info[CONF_MODEL]
+
+    from pyubee import Ubee
+    ubee = Ubee(host, username, password, model)
+    if not ubee.login():
+        _LOGGER.error("Login failed")
         return None
 
+    scanner = UbeeDeviceScanner(ubee)
+    return scanner
+
 
 class UbeeDeviceScanner(DeviceScanner):
     """This class queries a wireless Ubee router."""
 
-    def __init__(self, config):
+    def __init__(self, ubee):
         """Initialize the Ubee scanner."""
-        from pyubee import Ubee
-
-        self.host = config[CONF_HOST]
-        self.username = config[CONF_USERNAME]
-        self.password = config[CONF_PASSWORD]
-
-        self.last_results = {}
-        self.mac2name = {}
-
-        self.ubee = Ubee(self.host, self.username, self.password)
-        _LOGGER.info("Logging in")
-        results = self.get_connected_devices()
-        self.success_init = results is not None
-
-        if self.success_init:
-            self.last_results = results
-        else:
-            _LOGGER.error("Login failed")
+        self._ubee = ubee
+        self._mac2name = {}
```

Comment:
This dict is never populated, only accessed.

Reply:
Fixed.